File: services/api/src/atlas_api/models/output.py
"""Output artifact domain models — mirrors packages/shared/src/types/output.ts."""

from pydantic import BaseModel, ConfigDict

from atlas_api.models.enums import OutputArtifactKind


class OutputArtifact(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    kind: OutputArtifactKind
    title: str
    storage_key: str | None
    public_url: str | None
    evidence_pack_id: str | None
    generated_at: str
    model: str | None
    confidence_notes: str | None
    source_count: int
    created_at: str
    updated_at: str

File: packages/shared/src/types/output.ts
import type { OutputArtifactId, WorkspaceId, EvidencePackId } from "./ids.js";
import type { OutputArtifactKind } from "./enums.js";

export interface OutputArtifact {
  readonly id: OutputArtifactId;
  readonly workspaceId: WorkspaceId;
  readonly kind: OutputArtifactKind;
  readonly title: string;
  readonly storageKey: string | null;
  readonly publicUrl: string | null;
  readonly evidencePackId: EvidencePackId | null;
  readonly generatedAt: string;
  readonly model: string | null;
  readonly confidenceNotes: string | null;
  readonly sourceCount: number;
  readonly createdAt: string;
  readonly updatedAt: string;
}
